src/stream/src/executor/source/source_executor.rs

```diff
@@ -492,6 +499,15 @@ impl<S: StateStore> SourceExecutor<S> {
             }
         }
 
+        // Since DDL always accompany with a checkpoint barrier, we can assume that
+        // if initial_dispatch_num != 0, the source state must be initialized.
+        if self.actor_ctx.initial_dispatch_num != 0 {
+            assert!(
+                !is_uninitialized,
+                "is_uninitialized must be false if initial_dispatch_num != 0"
+            );
+        }
```

Comment:
I believe the assertion here is no-op.

Reply:
agree. `is_uninitialized` is inited with `self.actor_ctx.initial_dispatch_num == 0`